packages/flutter/lib/src/widgets/automatic_keep_alive.dart

```diff
@@ -220,10 +267,86 @@ class KeepAliveNotification extends Notification {
 ///
 /// Used with [KeepAliveNotification] objects as their
 /// [KeepAliveNotification.handle].
+///
+/// For a more convenient way to interact with [AutomaticKeepAlive] widgets,
+/// consider using [AutomaticKeepAliveClientMixin], which uses a
+/// [KeepAliveHandle] internally.
 class KeepAliveHandle extends ChangeNotifier {
   /// Trigger the listeners to indicate that the widget
   /// no longer needs to be kept alive.
   void release() {
     notifyListeners();
   }
 }
+
+/// A mixin with convenience methods for clients of [AutomaticKeepAlive].
+///
+/// Subclasses must implement [wantKeepAlive], and their [build] methods must
+/// call `super.build` (which will always return null).
+///
+/// Then, whenever [wantKeepAlive]'s value changes (or might change), the
+/// subclass should call [updateKeepAlive].
+///
+/// See also:
+///
+///  * [AutomaticKeepAlive], which listens to messages from this mixin.
+///  * [KeepAliveNotification], the notifications sent by this mixin.
+abstract class AutomaticKeepAliveClientMixin extends State<StatefulWidget> {
+  // This class is intended to be used as a mixin, and should not be
+  // extended directly.
+  factory AutomaticKeepAliveClientMixin._() => null;
```

Comment:
It's using three tricks at once: a private constructor makes the class unconstructable, a factory makes the private constructor not interfere with subclasses that mix this in, and the factory returning null is the shortest valid way of returning something that is type-correct. I added this to all the mixins at some point recently. (I didn't come up with the pattern, it was suggested to me in a discussion on some Dart SDK issue.)